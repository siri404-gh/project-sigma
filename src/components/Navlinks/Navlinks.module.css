.li {
  display: inline-block;
  margin: 0 20px;
}
