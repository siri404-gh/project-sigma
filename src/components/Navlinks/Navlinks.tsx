import React, { FC } from 'react'

import { Box, Link as MuiLink } from '@mui/material'
import { styled } from '@mui/system'
import Link from 'next/link'

export interface NavlinkRoute {
  title: string
  route: string
}
export interface NavlinkType {
  title: string
  url: string
  routes?: NavlinkRoute[]
}

export interface NavlinksProps {
  links?: NavlinkType[]
  rest?: any[]
}

const StyledBox = styled(Box)(({ theme }) => ({
  display: 'none',
  [theme.breakpoints.up('md')]: {
    display: 'block',
  },
  listStyle: 'none',
  padding: 0,
  margin: 0,
}))

const StyledLi = styled(Box)({
  display: 'inline-block',
  margin: '0 10px',
})

const Navlinks: FC<NavlinksProps> = ({ links = [], ...rest }) => (
  <StyledBox component='ul' {...rest}>
    {links.map(({ title, url }) => (
      <StyledLi key={title}>
        <Link href={url} passHref>
          <MuiLink underline='hover' variant='body2'>
            {title}
          </MuiLink>
        </Link>
      </StyledLi>
    ))}
  </StyledBox>
)
export default Navlinks
