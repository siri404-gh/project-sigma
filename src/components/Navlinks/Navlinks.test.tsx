import * as React from 'react'

import { render } from '@testing-library/react'

import Navlinks from '@/components/Navlinks/Navlinks'

describe('Navlinks', () => {
  test('renders', () => {
    const { container } = render(<Navlinks />)
    expect(container).toMatchSnapshot()
  })
})
