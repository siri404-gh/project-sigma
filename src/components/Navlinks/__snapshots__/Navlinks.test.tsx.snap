// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navlinks renders 1`] = `
<div>
  <ul
    class="MuiBox-root css-4s7zdz"
  />
</div>
`;
